.github/workflows/presubmit.yml

```diff
@@ -72,6 +72,28 @@ jobs:
             exit 1
           fi
 
+  check_clang_tidy:
+    name: Check clang-tidy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
+      - name: Get docker images
+        run: |
+          # https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-ghcrio
+          echo $CR_PAT | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+          docker pull ghcr.io/taichi-dev/taichidev-cpu-ubuntu18.04:v0.1.0
+        env:
+          CR_PAT: ${{ secrets.GITHUB_TOKEN }}
+      - name: Run clang-tidy
+        run: |
+          docker create --user root --name check_clang_tidy ghcr.io/taichi-dev/taichidev-cpu-ubuntu18.04:v0.1.0 /home/dev/taichi/.github/workflows/scripts/check_clang_tidy.sh "$CI_SETUP_CMAKE_ARGS"
```

Comment:
we need root user here because we need to apt install later? Then, it might be better to install the package in the image.

Reply:
Huh, actually only clang-tidy binary needs to be installed later. I didn't add to it the base image because all the other tests don't use it. WDYT?

Reply:
i guess one package is fine. The risk of allowing root user is higher. I think running clang-tidy should just like build/test on non-root mode.

Reply:
IMO all commands take effects only in the container. So I can't get what the risk of allowing root user is.

Reply:
a quick google search gives me these ;/

[Rootless mode allows running the Docker daemon and containers as a non-root user to mitigate potential vulnerabilities in the daemon and the container runtime.](https://docs.docker.com/engine/security/rootless/)

[Because anyone who accesses your container running as root can start undesirable processes in it, such as injecting malicious code. And running a process in your container as root makes it possible to change the user id (UID) or group id (GID) when starting the container, which makes your application vulnerable](https://engineering.bitnami.com/articles/why-non-root-containers-are-important-for-security.html#:~:text=Because%20anyone%20who%20accesses%20your,which%20makes%20your%20application%20vulnerable.)